zio-http/jvm/src/test/scala/zio/http/HybridStreamingSpec.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2021 - 2023 Sporta Technologies PVT LTD & the ZIO HTTP contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.http
+
+import zio._
+import zio.test.Assertion.equalTo
+import zio.test.TestAspect.{diagnose, sequential, shrinks, withLiveClock}
+import zio.test.{assertTrue, assertZIO}
+
+import zio.http.ServerSpec.requestBodySpec
+import zio.http.internal.{DynamicServer, HttpRunnableSpec}
+import zio.http.netty.NettyConfig
+
+object HybridRequestStreamingServerSpec extends HttpRunnableSpec {
```

Comment:
I think we need a few extra tests / tweeks in here:

1. We need to ensure that the pipeline is properly reset (and functional) after we changed the handler. We need to make multiple requests within the same test that reuses the same underlying connection, with bodies smaller and larger than the limit.
2. You need to use `body.isComplete` to check if the request body is streaming or not instead of `body.asStream`. For cases that the `size < maxSize` it should be `true`, or `false` when its larger than `maxSize`.